src/compiler/checker.ts

```diff
@@ -21275,6 +21275,10 @@ namespace ts {
                         (isBinaryExpression(target) && target.operatorToken.kind === SyntaxKind.CommaToken && isMatchingReference(source, target.right));
             }
             switch (source.kind) {
+                case SyntaxKind.MetaProperty:
+                    return target.kind === SyntaxKind.MetaProperty
+                        && (source as MetaProperty).keywordToken === SyntaxKind.ImportKeyword
+                        && (target as MetaProperty).keywordToken === SyntaxKind.ImportKeyword;

```

Comment:
You should also future-proof this and check whether right right side of the dot is `meta`.

Reply:
(or if not specifically `meta`, that they at least match)

Reply:
Actually, `new.target` and friends should _probably_ also be narrowable (being another meta property), do you should probably check just that both the `keywordToken` and `name` fields match.

Reply:
1. I added it, though it's not really required since `checkImportMetaProperty` forbids anything else anyway.  So maybe not include it after all?
2. Re `new.target`, I'll ask @RyanCavanaugh.

Reply:
We definitely both don't error on and strongly type `new.target` (albeit slightly inaccurately - the type doesn't include `undefined`), so....